lang/cs/Org.Apache.REEF.DistributedR/TaskRunner.cs

```diff
@@ -63,18 +65,24 @@ public void Start()
             this.toDriverThread.Start();
         }
 
-        public void Stop(bool wait = false)
+        internal void Stop()
         {
             Logr.Log(Level.Info, "Shutting down");
 
-            if (!wait)
+            Submit(ShutdownMsg.Factory());
+            this.cancelToken.WaitHandle.WaitOne(3000);
+
+            if (this.cancelToken.IsCancellationRequested)
+            {
+                this.toDriverThread.Join();
```

Comment:
No because the driver may have crashed which means we will wait for ever. There probably needs to be a heart beat so we shutdown when the heart beat stops or the last job is done.